File: Ejercicio 7/transporte.py
class Transporte:
    def __init__(self, combustible, capacidad, velocidad):
        self.combustible = combustible
        self.capacidad = capacidad
        self.velocidad = velocidad

    def mostrar_info(self):
        print("Combustible:", self.combustible)
        print("Capacidad:", self.capacidad)
        print("Velocidad:", self.velocidad)

    def mover(self):
        pass


class Terrestre(Transporte):
    def mover(self):
        print("Se desplaza por carretera")


class Aereo(Transporte):
    def mover(self):
        print("Vuela por el aire")


class Maritimo(Transporte):
    def mover(self):
        print("Navega en el agua")



t1 = Terrestre("Gasolina", 5, 120)
t2 = Aereo("Queroseno", 180, 800)
t3 = Maritimo("Diésel", 300, 60)

transportes = [t1, t2, t3]

for t in transportes:
    print("Tipo:", type(t).__name__)
    t.mostrar_info()
    t.mover()
    print("-------------------")

File: Ejercicio 7/transporte.java
class Transporte {
    String combustible;
    int capacidad;
    double velocidad;

    public Transporte(String combustible, int capacidad, double velocidad) {
        this.combustible = combustible;
        this.capacidad = capacidad;
        this.velocidad = velocidad;
    }

    public void mostrarInfo() {
        System.out.println("Combustible: " + combustible);
        System.out.println("Capacidad: " + capacidad);
        System.out.println("Velocidad: " + velocidad);
    }

    public void mover() {}
}


class Terrestre extends Transporte {
    public Terrestre(String combustible, int capacidad, double velocidad) {
        super(combustible, capacidad, velocidad);
    }

    public void mover() {
        System.out.println("Se desplaza por carretera");
    }
}


class Aereo extends Transporte {
    public Aereo(String combustible, int capacidad, double velocidad) {
        super(combustible, capacidad, velocidad);
    }

    public void mover() {
        System.out.println("Vuela por el aire");
    }
}


class Maritimo extends Transporte {
    public Maritimo(String combustible, int capacidad, double velocidad) {
        super(combustible, capacidad, velocidad);
    }

    public void mover() {
        System.out.println("Navega en el agua");
    }
}


public class Main {
    public static void main(String[] args) {

        Transporte t1 = new Terrestre("Gasolina", 5, 120);
        Transporte t2 = new Aereo("Queroseno", 180, 800);
        Transporte t3 = new Maritimo("Diésel", 300, 60);

        Transporte[] transportes = {t1, t2, t3};

        for (Transporte t : transportes) {
            System.out.println("Tipo: " + t.getClass().getSimpleName());
            t.mostrarInfo();
            t.mover();
            System.out.println("-------------------");
        }
    }
}
